packages/@aws-cdk/aws-sagemaker/lib/endpoint-config.ts

```diff
@@ -0,0 +1,313 @@
+    const endpointConfigArn = cdk.Stack.of(scope).formatArn({
+      service: 'sagemaker',
+      resource: 'endpoint-config',
+      resourceName: endpointConfigName,
+    });
+
+    class Import extends cdk.Resource implements IEndpointConfig {
+      public endpointConfigArn = endpointConfigArn;
+      public endpointConfigName = endpointConfigName;
+    }
+
+    return new Import(scope, id, {
+      environmentFromArn: endpointConfigArn,
+    });
+  }
+
+  /**
+   * The ARN of the endpoint configuration.
+   */
+  public readonly endpointConfigArn: string;
+  /**
+   * The name of the endpoint configuration.
+   */
+  public readonly endpointConfigName: string;
+
+  private readonly _instanceProductionVariants: { [key: string]: InstanceProductionVariant; } = {};
+
+  constructor(scope: Construct, id: string, props: EndpointConfigProps = {}) {
+    super(scope, id, {
+      physicalName: props.endpointConfigName,
+    });
+
+    (props.instanceProductionVariants || []).map(p => this.addInstanceProductionVariant(p));
+
+    // create the endpoint configuration resource
+    const endpointConfig = new CfnEndpointConfig(this, 'EndpointConfig', {
+      kmsKeyId: (props.encryptionKey) ? props.encryptionKey.keyArn : undefined,
+      endpointConfigName: this.physicalName,
+      productionVariants: cdk.Lazy.any({ produce: () => this.renderInstanceProductionVariants() }),
+    });
+    this.endpointConfigName = this.getResourceNameAttribute(endpointConfig.attrEndpointConfigName);
+    this.endpointConfigArn = this.getResourceArnAttribute(endpointConfig.ref, {
+      service: 'sagemaker',
+      resource: 'endpoint-config',
+      resourceName: this.physicalName,
+    });
+  }
+
+  /**
+   * Add production variant to the endpoint configuration.
+   *
+   * @param props The properties of a production variant to add.
+   */
+  public addInstanceProductionVariant(props: InstanceProductionVariantProps): void {
+    if (props.variantName in this._instanceProductionVariants) {
+      throw new Error(`There is already a Production Variant with name '${props.variantName}'`);
+    }
+    this.validateProps(props);
+    this._instanceProductionVariants[props.variantName] = {
+      acceleratorType: props.acceleratorType,
+      initialInstanceCount: props.initialInstanceCount || 1,
+      initialVariantWeight: props.initialVariantWeight || 1.0,
+      instanceType: props.instanceType || InstanceType.T2_MEDIUM,
+      modelName: props.model.modelName,
+      variantName: props.variantName,
+    };
+  }
+
+  /**
+   * Get instance production variants associated with endpoint configuration.
+   */
+  public get instanceProductionVariants(): InstanceProductionVariant[] {
+    return Object.values(this._instanceProductionVariants);
+  }
+
+  /**
+   * Find instance production variant based on variant name
+   * @param name Variant name from production variant
+   */
+  public findInstanceProductionVariant(name: string): InstanceProductionVariant {
```

Comment:
When/how will this public API be used? Might benefit from a readme example.

Reply:
This method is used by the upcoming `Endpoint` L2 construct to examine the variants configured on its associated `EndpointConfig` resource.